File: common_book/src/mfxsqbyd/java/com/intelligent/reader/adapter/BookRecommendAdapter.java
package com.intelligent.reader.adapter;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.ding.basic.bean.Book;
import com.ding.basic.bean.RecommendBean;
import com.intelligent.reader.R;


import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

/**
 * @author lijun Lee
 * @desc 推荐书籍
 * @mail jun_li@dingyuegroup.cn
 * @data 2018/3/6 17:11
 */

public class BookRecommendAdapter extends BaseAdapter {

    private List<RecommendBean> mList;

    @Override
    public int getCount() {
        return mList == null ? 0 : mList.size();
    }

    @Override
    public Object getItem(int position) {
        return null;
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder hodler;
        if (convertView == null) {
            convertView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_cover_recommend_layout, parent, false);
            hodler = new ViewHolder();
            hodler.book_img_iv = (ImageView) convertView.findViewById(R.id.book_img_iv);
            hodler.book_name_tv = (TextView) convertView.findViewById(R.id.book_name_tv);
            hodler.book_info_tv = (TextView) convertView.findViewById(R.id.book_info_tv);
            hodler.book_author_tv = (TextView) convertView.findViewById(R.id.book_author_tv);
            hodler.book_cover_status = (TextView) convertView.findViewById(R.id.book_cover_status);
            hodler.book_cover_category = (TextView) convertView.findViewById(R.id.book_cover_category);
            hodler.book_cover_nomore = (TextView) convertView.findViewById(R.id.cover_recomm_nomore);
            convertView.setTag(hodler);
        } else {
            hodler = (ViewHolder) convertView.getTag();
        }
        RecommendBean book = mList.get(position);
        Glide.with(parent.getContext().getApplicationContext()).load(book.getSourceImageUrl())
                .placeholder(net.lzbook.kit.R.drawable.icon_book_cover_default)
                .error(net.lzbook.kit.R.drawable.icon_book_cover_default)
                .diskCacheStrategy(DiskCacheStrategy.ALL).into(hodler.book_img_iv);
        hodler.book_name_tv.setText(book.getBookName());
        hodler.book_info_tv.setText(book.getDescription());
        hodler.book_author_tv.setText(book.getAuthorName());

        if (!TextUtils.isEmpty(book.getSubGenre())) {
            hodler.book_cover_category.setText(book.getSubGenre());
        } else {
            hodler.book_cover_category.setText("未分类");
        }

        if ("FINISH".equals(book.getSerialStatus())) {
            hodler.book_cover_status.setText(parent.getContext().getString(R.string.book_cover_state_writing));
        } else {
            hodler.book_cover_status.setText(parent.getContext().getString(R.string.book_cover_state_written));
        }

        if (position + 1 == getCount()) {
            hodler.book_cover_nomore.setVisibility(View.VISIBLE);
        } else {
            hodler.book_cover_nomore.setVisibility(View.GONE);
        }
        return convertView;
    }

    public void setData(List<RecommendBean> data) {
        this.mList = data;
        notifyDataSetChanged();
    }


    private static class ViewHolder {
        ImageView book_img_iv;
        TextView book_name_tv;
        TextView book_info_tv;
        TextView book_author_tv;
        TextView book_cover_status;
        TextView book_cover_category;
        TextView book_cover_nomore;
    }
}

File: common_book/src/qbmfrmxs/java/com/intelligent/reader/adapter/BookRecommendAdapter.kt
package com.intelligent.reader.adapter

import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.ding.basic.bean.RecommendBean
import com.intelligent.reader.R

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import net.lzbook.kit.utils.AppUtils
import java.text.MessageFormat

class BookRecommendAdapter : BaseAdapter() {

    private var recommends: List<RecommendBean>? = null

    override fun getCount(): Int {
        return if (recommends == null) 0 else recommends!!.size
    }

    override fun getItem(position: Int): Any? {
        return null
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View? {
        var contentView = convertView

        val viewHolder: ViewHolder

        if (contentView == null) {
            contentView = LayoutInflater.from(parent?.context).inflate(R.layout.item_book_end_recommend, parent, false)
            viewHolder = ViewHolder()
            viewHolder.img_recommend_cover = contentView?.findViewById(R.id.img_recommend_cover)
            viewHolder.txt_recommend_name = contentView?.findViewById(R.id.txt_recommend_name)
            viewHolder.txt_recommend_popularity = contentView?.findViewById(R.id.txt_recommend_popularity)
            contentView.tag = viewHolder
        } else {
            viewHolder = contentView.tag as ViewHolder
        }

        val recommend = recommends!![position]

        Glide.with(parent?.context?.applicationContext)
                .load(recommend.sourceImageUrl)
                .placeholder(R.drawable.common_book_cover_default_icon)
                .error(R.drawable.common_book_cover_default_icon)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .into(viewHolder.img_recommend_cover!!)

        viewHolder.txt_recommend_name?.text = recommend.bookName

        viewHolder.txt_recommend_popularity?.text = AppUtils.getCommonReadNums(recommend.uv)

        return contentView
    }

    fun setData(recommends: List<RecommendBean>) {
        this.recommends = recommends
    }


    private class ViewHolder {
        internal var img_recommend_cover: ImageView? = null
        internal var txt_recommend_name: TextView? = null
        internal var txt_recommend_popularity: TextView? = null
    }
}
